File: mobile/lib/widgets/category_sidebar.dart
import 'package:flutter/material.dart';

import '../theme/app_theme.dart';

class SidebarItem {
  final String slug;
  final String name;
  const SidebarItem({required this.slug, required this.name});
}

/// Vertical category rail matching src/components/ecommerce/CategorySidebar.js.
class CategorySidebar extends StatelessWidget {
  final List<SidebarItem> items;
  final String activeSlug;
  final ValueChanged<String> onSelect;

  const CategorySidebar({
    super.key,
    required this.items,
    required this.activeSlug,
    required this.onSelect,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 92,
      decoration: const BoxDecoration(border: Border(right: BorderSide(color: AppColors.divider))),
      child: Column(
        children: [
          const SizedBox(height: 20),
          Container(
            width: 48,
            height: 48,
            decoration: const BoxDecoration(shape: BoxShape.circle, color: AppColors.greenSoft),
            child: const Icon(Icons.storefront_rounded, color: AppColors.green, size: 20),
          ),
          const SizedBox(height: 8),
          Expanded(
            child: ListView(
              children: items.map((item) {
                final active = item.slug == activeSlug;
                return GestureDetector(
                  onTap: () => onSelect(item.slug),
                  child: Container(
                    padding: const EdgeInsets.symmetric(vertical: 12, horizontal: 4),
                    decoration: BoxDecoration(
                      border: Border(
                        bottom: BorderSide(
                          color: active ? AppColors.green : Colors.transparent,
                          width: 2.5,
                        ),
                      ),
                    ),
                    child: Text(
                      item.name,
                      textAlign: TextAlign.center,
                      style: TextStyle(
                        fontWeight: active ? FontWeight.w800 : FontWeight.w500,
                        color: active ? AppColors.green : AppColors.textPrimary,
                        fontSize: 12,
                      ),
                    ),
                  ),
                );
              }).toList(),
            ),
          ),
        ],
      ),
    );
  }
}

File: src/components/ecommerce/CategorySidebar.js
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import StorefrontRoundedIcon from "@mui/icons-material/StorefrontRounded";

export default function CategorySidebar({ items, activeSlug, onSelect }) {
  return (
    <Box
      sx={{
        width: 92,
        flexShrink: 0,
        borderRight: "1px solid #EEEEEE",
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        pt: 2.5,
        gap: 0.5,
      }}
    >
      <Box
        sx={{
          width: 48,
          height: 48,
          borderRadius: "50%",
          bgcolor: "primary.light",
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          mb: 1,
        }}
      >
        <StorefrontRoundedIcon sx={{ color: "primary.main" }} fontSize="small" />
      </Box>

      {items.map((item) => {
        const active = item.slug === activeSlug;
        return (
          <Box
            key={item.slug}
            onClick={() => onSelect(item.slug)}
            sx={{
              width: "100%",
              textAlign: "center",
              py: 1.25,
              px: 0.5,
              cursor: "pointer",
              borderBottom: active ? "2.5px solid" : "2.5px solid transparent",
              borderColor: active ? "primary.main" : "transparent",
            }}
          >
            <Typography
              variant="caption"
              sx={{
                fontWeight: active ? 800 : 500,
                color: active ? "primary.main" : "text.primary",
                lineHeight: 1.3,
                display: "block",
              }}
            >
              {item.name}
            </Typography>
          </Box>
        );
      })}
    </Box>
  );
}
